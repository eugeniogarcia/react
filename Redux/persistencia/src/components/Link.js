import React from 'react'
import PropTypes from 'prop-types'

const Link = ({ estado, children, enClick, filtro }) => (
  <button onClick={enClick} disabled={estado} style={{ marginLeft: "4px",}}>
    {children} ({filtro})
  </button>
);

Link.propTypes = {
  estado: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  enClick: PropTypes.func.isRequired,
};

export default Link
