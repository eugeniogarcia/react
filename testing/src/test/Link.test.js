// Link.react.test.js
import React from "react";
import Link from "../Link";
import renderer from "react-test-renderer";

describe('Testing basic component rendering', () => {
    test("renderer.create", () => {

        const component = renderer.create(<Link page="http://www.facebook.com">Facebook</Link>);

        /*Comprobación con snapshot
        Con esta técnica hacemos regression testing. Comprobamos que el layout
        no haya cambiado.
        La primera vez que se ejecuta el test se guardara un snapshot, se almacena
        en el directorio `__snapshots__`. En subsiguientes ejecuciones lo que hacemos es comparar con el snapshot previo
        */
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // manually trigger the callback
        tree.props.onMouseEnter();

        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // manually trigger the callback
        tree.props.onMouseLeave();
        
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("shallow", () => {

        const componente = shallow(<Link nombre="Eugenio" apellido="Garcia">
            <a href="www.elpais.com">El Pais</a>
            <br />
            <a href="www.spiegel.de">Der Spiegel</a>
            <br />
            <a href="www.sn.at">Salzburger</a>
        </Link>);

        expect(componente).toMatchSnapshot();
        // manually trigger the callback
        componente.simulate('mouseenter');

        expect(componente).toMatchSnapshot();

        // manually trigger the callback
        componente.simulate('mouseleave');
        expect(componente).toMatchSnapshot();
    });    

    test("mount", () => {

        const comp = mount(<Link nombre="Eugenio" apellido="Garcia" page="http://www.facebook.com">Facebook</Link>);

        expect(comp).toMatchSnapshot();

        // manually trigger the callback
        comp.find('[id="zona"]').simulate('mouseenter');

        expect(comp).toMatchSnapshot();

        // manually trigger the callback
        comp.find('div').simulate('mouseleave');
        expect(comp).toMatchSnapshot();
    });    
});

describe('Testing props', () => {

    test("props", () => {

        const comp = mount(<Link nombre="Eugenio" apellido="Garcia" page="http://www.facebook.com">Facebook</Link>);

        expect(comp.props().nombre).toEqual('Eugenio');
        comp.setProps({ nombre: 'Eugenio', apellido: 'Garcia Zach' });
        expect(comp.props().apellido).toEqual('Garcia Zach');

    });    

});