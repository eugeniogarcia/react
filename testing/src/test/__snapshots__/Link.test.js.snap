// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprueba que Link cambie cuando se pasae el ratonpor encima 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Comprueba que Link cambie cuando se pasae el ratonpor encima 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Comprueba que Link cambie cuando se pasae el ratonpor encima 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
